### DISCUSSION TITLE
Trying to set x-limits on my posterior plot

### DISCUSSION THREAD

**Discussion by sohinixcvii** [Samreay/ChainConsumer#115] [Q&A] [answered]
Hi folks!

I have been trying to use ChainConsumer to plot the posterior from my MCMC and it looks like adding the 'extents' argument does nothing for my plot. I am also unable to import PlotConfig from ChainConsumer. Any help would be appreciated. Details below:

**Versions:**
chainconsumer v1.0.2
python v3.8.18

**Code:**

````
from chainconsumer import ChainConsumer

i=50

data=np.loadtxt('pk'+str(i)+'.log') #sample saved with name 'model num.out'
params_test=np.loadtxt('data/params_LH.txt')
truth = [params_test[i,0]]

c = ChainConsumer()
c.add_chain(data[:,0], parameters=[r"$M_{(h, {\rm min})}(\rm 10^{10} M_\odot)$"],
                name='CV+Noise',color='#F28482',)
c.configure(label_font_size=18,linestyles='-',linewidths=2, tick_font_size=18,shade_alpha=1 )
c.configure_truth(color='k', ls=":", lw=1.5)
fig = c.plotter.plot(truth=truth,extents={"$M_{(h, {\rm min})}(\rm 10^{10} M_\odot)$": (2.0, 8)})
fig.set_size_inches(5 + fig.get_size_inches())
````

![plot_ind_50_mhmin_3 73_fact_4](https://github.com/Samreay/ChainConsumer/assets/66513657/14f6c343-f834-4d4c-a12e-6c001ae11f8d)
[pk50.log](https://github.com/Samreay/ChainConsumer/files/14011368/pk50.log)
[params_LH.txt](https://github.com/Samreay/ChainConsumer/files/14011377/params_LH.txt)

**Comment by Samreay** [comment, marked as answer]
So I did find an issue with the latest version of Chainconsumer in this, but it wasn't to do with the extents. Also, you may have your environment misconfigured, because the code you provided only works on ChainConsumer <1 (as you note with the failing import)

If you update to 1.0.3, this code should give you what you want:

```python
import pandas as pd

from chainconsumer import Chain, ChainConsumer, PlotConfig, Truth

df = pd.read_csv("pk50.log", sep=r"\s+", header=None, names=["mass", "something"])
chain = Chain(samples=df[["mass"]], name="CV+noise", color="#F28482")

df_truth = pd.read_csv("params_LH.txt", sep=r"\s+", header=None, names=["mass", "something"])
truth = Truth(location={"mass": df_truth["mass"].iloc[50]})

config = PlotConfig(labels={"mass": r"$M_{(h, {\rm min})}(\rm 10^{10} M_\odot)$"}, extents={"mass": (2, 8)})

c = ChainConsumer()
c.add_chain(chain)
c.add_truth(truth)
c.set_plot_config(config)

fig = c.plotter.plot(figsize=(6, 4), filename="tmp.png")
```

Also, if you want the bars to look smoother when zoomed in, try adding `smooth=10` to the `Chain` kwargs :)

![tmp](https://github.com/Samreay/ChainConsumer/assets/6748505/2de1e505-046c-4e51-81db-3b6e2aebffc5)

**Reply by sohinixcvii** [reply]
Oh wow thank you so much!! This is exactly what I wanted! And yes, I was building off of a project written years ago, didn't realise the changes. 😬
Thanks again!! 🥹🥹
